src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,46 +1,94 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  handleCloseModal: () => void;
+  loading: boolean;
+  setLoading: (con: boolean) => void;
+  setModalVisible: (con: boolean) => void;
+  selectedTodo: Todo;
+};
+
+export const TodoModal: React.FC<Props> = (
+  {
+    handleCloseModal,
+    loading,
+    setLoading,
+    setModalVisible,
+    selectedTodo,
+  },
+) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      setLoading(false);
+    }, 500);
+
+    return () => {
+      clearTimeout(timeout);
+      setModalVisible(false);
+    };
+  }, [setLoading, setModalVisible]);
+
+  useEffect(() => {
+    if (selectedTodo && selectedTodo.userId) {
+      getUser(selectedTodo.userId)
+        .then(setUser)
+        .catch((error) => {
```

Comment:
this catch block is redundant - you just throw the same error and don't handle it - that will cause page crush. Handle the error (in similar way as in App.tsx or remove it at all)